smarts/sstudio/types.py

```diff
@@ -434,6 +486,31 @@ class Traffic:
     flows: Sequence[Flow]
     """Flows are used to define a steady supply of vehicles."""
 
+    single_vehicles: Sequence[SingleVehicle] = field(default_factory=list)
+    """Single instance vehicles"""
+
+    @cached_property
+    def actors(self) -> Sequence[TrafficActor]:
+        actors = set()
+        for a in [a for flow in self.flows for a in flow.actors]:
+            actors.add(a)
+
+        for a in [a for sv in self.single_vehicles for a in sv.actors]:
+            actors.add(a)
```

Comment:
It seems this can be done more efficiently with set comprehensions:
```python
    return {a for flow in self.flows for a in flow.actors} | {a for sv in self.single_vehicles for a in sv.actors}
```